Makefile

```diff
@@ -12,13 +12,13 @@ test:
 
 .PHONY: pre-commit
 pre-commit:
-	./utils/stylua --config-path stylua.toml --glob lua/**/*.lua -- lua
+	./utils/stylua --config-path stylua.toml --glob 'lua/**/*.lua' -- lua
 	luacheck lua
 	vusted lua
 
 .PHONY: integration
 integration:
-	./utils/stylua --config-path stylua.toml --check --glob lua/**/*.lua -- lua
```

Comment:
## In the invalid (` --glob lua/**/*.lua`) case

```bash
> bash -c "./utils/stylua --config-path stylua.toml --check --glob lua/**/*.lua -- lua"
```

## In the valid (` --glob 'lua/**/*.lua'`) case

```bash
> bash -c "./utils/stylua --config-path stylua.toml --check --glob 'lua/**/*.lua' -- lua"
Diff in lua/cmp/types/vim.lua:
15  15   | ---@class vim.Range
16  16   | ---@field public start vim.Position
17  17   | ---@field public end vim.Position
18       |-
Diff in lua/cmp/types/init.lua:
5   5    | types.vim = require('cmp.types.vim')
6   6    |
7   7    | return types
8        |-
Diff in lua/cmp/types/cmp.lua:
104 104  | ---@field public max_item_count number
105 105  |
106 106  | return cmp
107      |-
Diff in lua/cmp/types/lsp_spec.lua:
1        |-local spec = require'cmp.utils.spec'
2        |-local lsp = require'cmp.types.lsp'
    1    |+local spec = require('cmp.utils.spec')
    2    |+local lsp = require('cmp.types.lsp')
3   3    |
4        |-describe('types.lsp', function ()
    4    |+describe('types.lsp', function()
5   5    |   before_each(spec.before)
6        |-  describe('Position', function ()
    6    |+  describe('Position', function()
7   7    |     vim.fn.setline('1', {
8   8    |       'あいうえお',
9   9    |       'かきくけこ',
--------------------------------------------------------------------------------
44  44   |     assert.are.equal(lsp_position.character, 5)
45  45   |   end)
46  46   | end)
47       |-
Diff in lua/cmp/config/compare.lua:
1        |-local types = require'cmp.types'
2        |-local misc  = require 'cmp.utils.misc'
    1    |+local types = require('cmp.types')
    2    |+local misc = require('cmp.utils.misc')
3   3    |
4   4    | local compare = {}
5   5    |
--------------------------------------------------------------------------------
85  85   | end
86  86   |
87  87   | return compare
88       |-
Diff in lua/cmp/config/default.lua:
15  15   |       keyword_length = 1,
16  16   |       get_trigger_characters = function(trigger_characters)
17  17   |         return trigger_characters
18       |-      end
    18   |+      end,
19  19   |     },
20  20   |
21  21   |     snippet = {
--------------------------------------------------------------------------------
37  37   |       default_behavior = types.cmp.ConfirmBehavior.Insert,
38  38   |       get_commit_characters = function(commit_characters)
39  39   |         return commit_characters
40       |-      end
    40   |+      end,
41  41   |     },
42  42   |
43  43   |     sorting = {
--------------------------------------------------------------------------------
50  50   |         compare.sort_text,
51  51   |         compare.length,
52  52   |         compare.order,
53       |-      }
    53   |+      },
54  54   |     },
55  55   |
56  56   |     event = {},
--------------------------------------------------------------------------------
61  61   |       deprecated = true,
62  62   |       format = function(_, vim_item)
63  63   |         return vim_item
64       |-      end
    64   |+      end,
65  65   |     },
66  66   |
67  67   |     experimental = {
--------------------------------------------------------------------------------
71  71   |     sources = {},
72  72   |   }
73  73   | end
74       |-
Diff in lua/cmp/types/lsp.lua:
1        |-
2        |-local misc = require "cmp.utils.misc"
    1    |+local misc = require('cmp.utils.misc')
3   2    | ---@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/
4   3    | ---@class lsp
5   4    | local lsp = {}
--------------------------------------------------------------------------------
18  17   |   if #lines > 0 then
19  18   |     return {
20  19   |       row = position.line + 1,
21       |-      col = misc.to_vimindex(lines[1], position.character)
    20   |+      col = misc.to_vimindex(lines[1], position.character),
22  21   |     }
23  22   |   end
24  23   |   return {
--------------------------------------------------------------------------------
196 195  | ---@field public dup boolean|nil
197 196  |
198 197  | return lsp
199      |-
Diff in lua/cmp/utils/async.lua:
27  27   |       timer:stop()
28  28   |
29  29   |       local delta = math.max(0, self.timeout - (vim.loop.now() - time))
30       |-      timer:start(delta, 0, vim.schedule_wrap(function()
31       |-        time = nil
32       |-        fn(unpack(args))
33       |-      end))
34       |-    end
    30   |+      timer:start(
    31   |+        delta,
    32   |+        0,
    33   |+        vim.schedule_wrap(function()
    34   |+          time = nil
    35   |+          fn(unpack(args))
    36   |+        end)
    37   |+      )
    38   |+    end,
35  39   |   })
36  40   | end
37  41   |
--------------------------------------------------------------------------------
52  56   | end
53  57   |
54  58   | return async
55       |-
Diff in lua/cmp/utils/pattern.lua:
26  26   | end
27  27   |
28  28   | return pattern
29       |-
Diff in lua/cmp/utils/spec.lua:
1        |-local context = require'cmp.context'
2        |-local source  = require 'cmp.source'
    1    |+local context = require('cmp.context')
    2    |+local source = require('cmp.source')
3   3    | local types = require('cmp.types')
4   4    |
5   5    | local spec = {}
6   6    |
7   7    | spec.before = function()
8        |-  vim.cmd [[
    8    |+  vim.cmd([[
9   9    |     bdelete!
10  10   |     enew!
11  11   |     setlocal virtualedit=all
12       |-  ]]
    12   |+  ]])
13  13   | end
14  14   |
15  15   | spec.state = function(text, row, col)
--------------------------------------------------------------------------------
17  17   |   vim.fn.cursor(row, col)
18  18   |   local ctx = context.empty()
19  19   |   local s = source.new('spec', {
20       |-    complete = function()
21       |-    end
    20   |+    complete = function() end,
22  21   |   })
23  22   |   return {
24  23   |     context = function()
--------------------------------------------------------------------------------
46  45   |       ctx = context.new(ctx, { reason = types.cmp.ContextReason.Manual })
47  46   |       s:complete(ctx, function() end)
48  47   |       return ctx
49       |-    end
    48   |+    end,
50  49   |   }
51  50   | end
52  51   |
53  52   | return spec
54       |-
Diff in lua/cmp/utils/char.lua:
110 110  | end
111 111  |
112 112  | return char
113      |-
Diff in lua/cmp/utils/misc_spec.lua:
8   8    |   it('merge', function()
9   9    |     local merged
10  10   |     merged = misc.merge({
11       |-      a = {}
    11   |+      a = {},
12  12   |     }, {
13  13   |       a = {
14       |-        b = 1
15       |-      }
    14   |+        b = 1,
    15   |+      },
16  16   |     })
17  17   |     assert.are.equal(merged.a.b, 1)
18  18   |     merged = misc.merge({
19       |-      a = false
    19   |+      a = false,
20  20   |     }, {
21  21   |       a = {
22       |-        b = 1
23       |-      }
    22   |+        b = 1,
    23   |+      },
24  24   |     })
25  25   |     assert.are.equal(merged.a, false)
26  26   |   end)
27  27   | end)
28       |-
Diff in lua/cmp/utils/cache.lua:
54  54   | end
55  55   |
56  56   | return cache
57       |-
Diff in lua/cmp/utils/async_spec.lua:
1        |-local async = require "cmp.utils.async"
    1    |+local async = require('cmp.utils.async')
2   2    |
3   3    | describe('utils.async', function()
4        |-
5   4    |   it('throttle', function()
6   5    |     local count = 0
7   6    |     local now
--------------------------------------------------------------------------------
13  12   |     now = vim.loop.now()
14  13   |     f.timeout = 100
15  14   |     f()
16       |-    vim.wait(1000, function() return count == 1 end)
    15   |+    vim.wait(1000, function()
    16   |+      return count == 1
    17   |+    end)
17  18   |     assert.is.truthy(math.abs(f.timeout - (vim.loop.now() - now)) < 10)
18  19   |
19  20   |     -- 2. delay for 500ms
20  21   |     now = vim.loop.now()
21  22   |     f.timeout = 500
22  23   |     f()
23       |-    vim.wait(1000, function() return count == 2 end)
    24   |+    vim.wait(1000, function()
    25   |+      return count == 2
    26   |+    end)
24  27   |     assert.is.truthy(math.abs(f.timeout - (vim.loop.now() - now)) < 10)
25  28   |
26  29   |     -- 4. delay for 500ms and wait 100ms (remain 400ms)
--------------------------------------------------------------------------------
32  35   |     now = vim.loop.now()
33  36   |     f.timeout = 100
34  37   |     f()
35       |-    vim.wait(1000, function() return count == 3 end)
    38   |+    vim.wait(1000, function()
    39   |+      return count == 3
    40   |+    end)
36  41   |     assert.is.truthy(math.abs(vim.loop.now() - now) < 10)
37  42   |   end)
38       |-
39  43   | end)
40       |-
Diff in lua/cmp/utils/str_spec.lua:
1        |-local str = require "cmp.utils.str"
    1    |+local str = require('cmp.utils.str')
2   2    |
3   3    | describe('utils.str', function()
4        |-
5   4    |   it('get_word', function()
6   5    |     assert.are.equal(str.get_word('print'), 'print')
7   6    |     assert.are.equal(str.get_word('$variable'), '$variable')
--------------------------------------------------------------------------------
27  26   |     assert.are.equal(str.escape('plain\\"', {}), 'plain\\\\"')
28  27   |     assert.are.equal(str.escape('pla"in', { '"' }), 'pla\\"in')
29  28   |   end)
30       |-
31  29   | end)
32       |-
33       |-
Diff in lua/cmp/utils/misc.lua:
21  21   | ---@param v2 T
22  22   | ---@return T
23  23   | misc.merge = function(v1, v2)
24       |-  local merge1 = type(v1) == "table" and (not vim.tbl_islist(v1) or vim.tbl_isempty(v1))
25       |-  local merge2 = type(v2) == "table" and (not vim.tbl_islist(v1) or vim.tbl_isempty(v1))
    24   |+  local merge1 = type(v1) == 'table' and (not vim.tbl_islist(v1) or vim.tbl_isempty(v1))
    25   |+  local merge2 = type(v2) == 'table' and (not vim.tbl_islist(v1) or vim.tbl_isempty(v1))
26  26   |   if merge1 and merge2 then
27  27   |     local new_tbl = {}
28  28   |     for k, v in pairs(v2) do
--------------------------------------------------------------------------------
47  47   |
48  48   |   return v1
49  49   | end
50       |-
51  50   |
52  51   | ---Generate id for group name
53  52   | misc.id = setmetatable({
54       |-  group = {}
    53   |+  group = {},
55  54   | }, {
56  55   |   __call = function(_, group)
57  56   |     misc.id.group[group] = misc.id.group[group] or 0
58  57   |     misc.id.group[group] = misc.id.group[group] + 1
59  58   |     return misc.id.group[group]
60       |-  end
    59   |+  end,
61  60   | })
62  61   |
63  62   | ---Check the value is nil or not.
Diff in lua/cmp/utils/str.lua:
1        |-local char = require'cmp.utils.char'
2        |-local pattern = require 'cmp.utils.pattern'
    1    |+local char = require('cmp.utils.char')
    2    |+local pattern = require('cmp.utils.pattern')
3   3    |
4   4    | local str = {}
5   5    |
--------------------------------------------------------------------------------
179 179  | end
180 180  |
181 181  | return str
182      |-
Diff in lua/cmp/utils/keymap.lua:
26  26   |   while i <= #keys do
27  27   |     if string.sub(keys, i, i) == '<' then
28  28   |       if not vim.tbl_contains({ '<lt>', '<Lt>', '<lT>', '<LT>' }, string.sub(keys, i, i + 3)) then
29       |-        keys = string.sub(keys, 1, i -1) .. '<LT>' ..  string.sub(keys, i + 1)
    29   |+        keys = string.sub(keys, 1, i - 1) .. '<LT>' .. string.sub(keys, i + 1)
30  30   |         i = i + 3
31  31   |       end
32  32   |     end
--------------------------------------------------------------------------------
80  80   |
81  81   | ---Feedkeys with callback
82  82   | keymap.feedkeys = setmetatable({
83       |-  callbacks = {}
    83   |+  callbacks = {},
84  84   | }, {
85  85   |   __call = function(self, keys, mode, callback)
86  86   |     if #keys ~= 0 then
--------------------------------------------------------------------------------
104 104  |         wait()
105 105  |       end
106 106  |     end
107      |-  end
    107  |+  end,
108 108  | })
109 109  | misc.set(_G, { 'cmp', 'utils', 'keymap', 'feedkeys', 'run' }, function(id)
110 110  |   if keymap.feedkeys.callbacks[id] then
--------------------------------------------------------------------------------
201 201  | end)
202 202  |
203 203  | return keymap
204      |-
```